packages/react-form/src/hooks/test/form-with-dynamic-list.test.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import {mount} from '@shopify/react-testing';
+
+import {
+  FormWithDynamicVariantList,
+  TextField,
+  isDirty,
+  fakeProduct,
+  hitSubmit,
+  hitReset,
+  waitForSubmit,
+} from './utilities';
+
+import {submitSuccess, submitFail} from '..';
+
+describe('useForm with dynamic list', () => {
```

Comment:
those tests are inspired by origin `useForm` test. 

`makeClean` and `makeCleanAfterSubmit` seem broken with dynamic lists 🤔